### PR TITLE
Standardize GCloud integration's logging

### LINKED ISSUE #10334: `wazuh-modulesd:gcp-pubsub` logging is not consistent
|Wazuh version|Component|Install type|
|---|---|---|
| 4.2.2 | Wazuh modulesd | Manager | 

## Description

After analyzing the logging from the `gcp-pubsub` integration through `modulesd` we noticed some inconsistencies. Depending on the logging configuration, there are logs that may not appear even though they match the logging level. One of these messages is key for the QA integration tests: _Received and acknowledged X messages_

### modulesd.debug=2 | gcp-logging=info

```
2021/09/28 13:48:18 wazuh-modulesd[1854] main.c:94 at main(): DEBUG: Created new thread for the 'gcp-pubsub' module.
2021/09/28 13:48:18 wazuh-modulesd:gcp-pubsub[1854] wm_gcp.c:48 at wm_gcp_main(): INFO: Module started.
2021/09/28 13:48:18 wazuh-modulesd:gcp-pubsub[1854] wm_gcp.c:64 at wm_gcp_main(): DEBUG: Starting fetching of logs.
2021/09/28 13:48:18 wazuh-modulesd:gcp-pubsub[1854] wm_gcp.c:87 at wm_gcp_run(): DEBUG: Create argument list
2021/09/28 13:48:18 wazuh-modulesd:gcp-pubsub[1854] wm_gcp.c:137 at wm_gcp_run(): DEBUG: Launching command: wodles/gcloud/gcloud --project wazuh-dev-258815 --subscription_id test_gcloud_blog --credentials_file /var/ossec/wodles/wazuh-dev.json --max_messages 5 --num_threads 1 --log_level 2
2021/09/28 13:48:20 wazuh-modulesd:syscollector[1854] wm_syscollector.c:87 at wm_sys_log(): DEBUG: Sync sent: {"component":"syscollector_processes","data":{"attributes":{"argvs":"--color=auto gcp","checksum":"5c140a48e7c451b7ad86e0cac18b33a9faeaedf0","cmd":"grep","egroup":"root","euser":"root","fgroup":"root","name":"grep","nice":0,"nlwp":1,"pgrp":1200,"pid":"1201","ppid":71,"priority":20,"processor":4,"resident":185,"rgroup":"root","ruser":"root","scan_time":"2021/09/28 13:48:20","session":71,"sgroup":"root","share":160,"size":828,"start_time":2512991,"state":"S","stime":1,"suser":"root","tgid":1201,"tty":34817,"utime":0,"vm_size":3312},"index":"1201","timestamp":""},"type":"state"}
2021/09/28 13:48:20 wazuh-db[1522] wdb_parser.c:250 at wdb_parse(): DEBUG: Agent 000 query: syscollector_processes save2 {"attributes":{"argvs":"--color=auto gcp","checksum":"5c140a48e7c451b7ad86e0cac18b33a9faeaedf0","cmd":"grep","egroup":"root","euser":"root","fgroup":"root","name":"grep","nice":0,"nlwp":1,"pgrp":1200,"pid":"1201","ppid":71,"priority":20,"processor":4,"resident":185,"rgroup":"root","ruser":"root","scan_time":"2021/09/28 13:48:20","session":71,"sgroup":"root","share":160,"size":828,"start_time":2512991,"state":"S","stime":1,"suser":"root","tgid":1201,"tty":34817,"utime":0,"vm_size":3312},"index":"1201","timestamp":""}
2021/09/28 13:48:39 wazuh-modulesd:gcp-pubsub[1854] wm_gcp.c:68 at wm_gcp_main(): DEBUG: Fetching logs finished.
2021/09/28 13:48:39 wazuh-modulesd:gcp-pubsub[1854] wm_gcp.c:60 at wm_gcp_main(): DEBUG: Sleeping until: 2021/09/28 13:49:18
```

### modulesd.debug=2 | gcp-logging=debug 

```
2021/09/28 13:50:27 wazuh-modulesd[3146] main.c:94 at main(): DEBUG: Created new thread for the 'gcp-pubsub' module.
2021/09/28 13:50:27 wazuh-modulesd:gcp-pubsub[3146] wm_gcp.c:48 at wm_gcp_main(): INFO: Module started.
2021/09/28 13:50:27 wazuh-modulesd:gcp-pubsub[3146] wm_gcp.c:64 at wm_gcp_main(): DEBUG: Starting fetching of logs.
2021/09/28 13:50:27 wazuh-modulesd:gcp-pubsub[3146] wm_gcp.c:87 at wm_gcp_run(): DEBUG: Create argument list
2021/09/28 13:50:27 wazuh-modulesd:gcp-pubsub[3146] wm_gcp.c:137 at wm_gcp_run(): DEBUG: Launching command: wodles/gcloud/gcloud --project wazuh-dev-258815 --subscription_id test_gcloud_blog --credentials_file /var/ossec/wodles/wazuh-dev.json --max_messages 5 --num_threads 1 --log_level 1
2021/09/28 13:50:29 wazuh-modulesd:syscollector[3146] wm_syscollector.c:87 at wm_sys_log(): DEBUG: Sync sent: {"component":"syscollector_processes","data":{"attributes":{"argvs":"--color=auto gcp","checksum":"24b95d4f833f63fbad9456a78d8dcbaa46577989","cmd":"grep","egroup":"root","euser":"root","fgroup":"root","name":"grep","nice":0,"nlwp":1,"pgrp":1200,"pid":"1201","ppid":71,"priority":20,"processor":1,"resident":185,"rgroup":"root","ruser":"root","scan_time":"2021/09/28 13:50:29","session":71,"sgroup":"root","share":160,"size":828,"start_time":2512991,"state":"S","stime":5,"suser":"root","tgid":1201,"tty":34817,"utime":3,"vm_size":3312},"index":"1201","timestamp":""},"type":"state"}
2021/09/28 13:50:29 wazuh-db[2907] wdb_parser.c:250 at wdb_parse(): DEBUG: Agent 000 query: syscollector_processes save2 {"attributes":{"argvs":"--color=auto gcp","checksum":"24b95d4f833f63fbad9456a78d8dcbaa46577989","cmd":"grep","egroup":"root","euser":"root","fgroup":"root","name":"grep","nice":0,"nlwp":1,"pgrp":1200,"pid":"1201","ppid":71,"priority":20,"processor":1,"resident":185,"rgroup":"root","ruser":"root","scan_time":"2021/09/28 13:50:29","session":71,"sgroup":"root","share":160,"size":828,"start_time":2512991,"state":"S","stime":5,"suser":"root","tgid":1201,"tty":34817,"utime":3,"vm_size":3312},"index":"1201","timestamp":""}
2021/09/28 13:50:49 wazuh-modulesd:gcp-pubsub[3146] wm_gcp.c:190 at wm_gcp_run(): DEBUG: gcloud_wodle: DEBUG: Setting 1 thread to pull 5 messages each
2021/09/28 13:50:49 wazuh-modulesd:gcp-pubsub[3146] wm_gcp.c:190 at wm_gcp_run(): DEBUG: gcloud_wodle: INFO: Received and acknowledged 0 messages
2021/09/28 13:50:49 wazuh-modulesd:gcp-pubsub[3146] wm_gcp.c:68 at wm_gcp_main(): DEBUG: Fetching logs finished.
2021/09/28 13:50:49 wazuh-modulesd:gcp-pubsub[3146] wm_gcp.c:60 at wm_gcp_main(): DEBUG: Sleeping until: 2021/09/28 13:51:27
```

### modulesd.debug=2 | gcp-logging=info

```
2021/09/28 13:54:14 wazuh-modulesd[4564] main.c:94 at main(): DEBUG: Created new thread for the 'gcp-pubsub' module.
2021/09/28 13:54:14 wazuh-modulesd:gcp-pubsub[4564] wm_gcp.c:48 at wm_gcp_main(): INFO: Module started.
2021/09/28 13:54:14 wazuh-modulesd:gcp-pubsub[4564] wm_gcp.c:64 at wm_gcp_main(): DEBUG: Starting fetching of logs.
2021/09/28 13:54:14 wazuh-modulesd:gcp-pubsub[4564] wm_gcp.c:87 at wm_gcp_run(): DEBUG: Create argument list
2021/09/28 13:54:14 wazuh-modulesd:gcp-pubsub[4564] wm_gcp.c:137 at wm_gcp_run(): DEBUG: Launching command: wodles/gcloud/gcloud --project wazuh-dev-258815 --subscription_id test_gcloud_blog --credentials_file /var/ossec/wodles/wazuh-dev.json --max_messages 5 --num_threads 1 --log_level 2
2021/09/28 13:54:16 wazuh-modulesd:syscollector[4564] wm_syscollector.c:87 at wm_sys_log(): DEBUG: Sync sent: {"component":"syscollector_processes","data":{"attributes":{"argvs":"--color=auto gcp","checksum":"00e0a0bda322ecbc690e3ecdb24bdcef8fa42789","cmd":"grep","egroup":"root","euser":"root","fgroup":"root","name":"grep","nice":0,"nlwp":1,"pgrp":1200,"pid":"1201","ppid":71,"priority":20,"processor":4,"resident":185,"rgroup":"root","ruser":"root","scan_time":"2021/09/28 13:54:16","session":71,"sgroup":"root","share":160,"size":828,"start_time":2512991,"state":"S","stime":10,"suser":"root","tgid":1201,"tty":34817,"utime":5,"vm_size":3312},"index":"1201","timestamp":""},"type":"state"}
2021/09/28 13:54:16 wazuh-db[4292] wdb_parser.c:250 at wdb_parse(): DEBUG: Agent 000 query: syscollector_processes save2 {"attributes":{"argvs":"--color=auto gcp","checksum":"00e0a0bda322ecbc690e3ecdb24bdcef8fa42789","cmd":"grep","egroup":"root","euser":"root","fgroup":"root","name":"grep","nice":0,"nlwp":1,"pgrp":1200,"pid":"1201","ppid":71,"priority":20,"processor":4,"resident":185,"rgroup":"root","ruser":"root","scan_time":"2021/09/28 13:54:16","session":71,"sgroup":"root","share":160,"size":828,"start_time":2512991,"state":"S","stime":10,"suser":"root","tgid":1201,"tty":34817,"utime":5,"vm_size":3312},"index":"1201","timestamp":""}
2021/09/28 13:54:35 wazuh-modulesd:gcp-pubsub[4564] wm_gcp.c:195 at wm_gcp_run(): INFO: - INFO - Received and acknowledged 0 messages
2021/09/28 13:54:35 wazuh-modulesd:gcp-pubsub[4564] wm_gcp.c:68 at wm_gcp_main(): DEBUG: Fetching logs finished.
2021/09/28 13:54:35 wazuh-modulesd:gcp-pubsub[4564] wm_gcp.c:60 at wm_gcp_main(): DEBUG: Sleeping until: 2021/09/28 13:55:14
```

### modulesd.debug=2 | gcp-logging=debug

```
2021/09/28 13:55:23 wazuh-modulesd[5945] main.c:94 at main(): DEBUG: Created new thread for the 'gcp-pubsub' module.
2021/09/28 13:55:23 wazuh-modulesd:gcp-pubsub[5945] wm_gcp.c:48 at wm_gcp_main(): INFO: Module started.
2021/09/28 13:55:23 wazuh-modulesd:gcp-pubsub[5945] wm_gcp.c:64 at wm_gcp_main(): DEBUG: Starting fetching of logs.
2021/09/28 13:55:23 wazuh-modulesd:gcp-pubsub[5945] wm_gcp.c:87 at wm_gcp_run(): DEBUG: Create argument list
2021/09/28 13:55:23 wazuh-modulesd:gcp-pubsub[5945] wm_gcp.c:137 at wm_gcp_run(): DEBUG: Launching command: wodles/gcloud/gcloud --project wazuh-dev-258815 --subscription_id test_gcloud_blog --credentials_file /var/ossec/wodles/wazuh-dev.json --max_messages 5 --num_threads 1 --log_level 1
2021/09/28 13:55:25 wazuh-modulesd:syscollector[5945] wm_syscollector.c:87 at wm_sys_log(): DEBUG: Sync sent: {"component":"syscollector_processes","data":{"attributes":{"argvs":"--color=auto gcp","checksum":"3865123b48b41e28929e869317665376be4a0474","cmd":"grep","egroup":"root","euser":"root","fgroup":"root","name":"grep","nice":0,"nlwp":1,"pgrp":1200,"pid":"1201","ppid":71,"priority":20,"processor":4,"resident":185,"rgroup":"root","ruser":"root","scan_time":"2021/09/28 13:55:25","session":71,"sgroup":"root","share":160,"size":828,"start_time":2512991,"state":"S","stime":15,"suser":"root","tgid":1201,"tty":34817,"utime":7,"vm_size":3312},"index":"1201","timestamp":""},"type":"state"}
2021/09/28 13:55:25 wazuh-db[5670] wdb_parser.c:250 at wdb_parse(): DEBUG: Agent 000 query: syscollector_processes save2 {"attributes":{"argvs":"--color=auto gcp","checksum":"3865123b48b41e28929e869317665376be4a0474","cmd":"grep","egroup":"root","euser":"root","fgroup":"root","name":"grep","nice":0,"nlwp":1,"pgrp":1200,"pid":"1201","ppid":71,"priority":20,"processor":4,"resident":185,"rgroup":"root","ruser":"root","scan_time":"2021/09/28 13:55:25","session":71,"sgroup":"root","share":160,"size":828,"start_time":2512991,"state":"S","stime":15,"suser":"root","tgid":1201,"tty":34817,"utime":7,"vm_size":3312},"index":"1201","timestamp":""}
2021/09/28 13:55:43 wazuh-modulesd:gcp-pubsub[5945] wm_gcp.c:187 at wm_gcp_run(): DEBUG: gcloud_wodle - DEBUG - Setting 1 thread to pull 5 messages each
2021/09/28 13:55:43 wazuh-modulesd:gcp-pubsub[5945] wm_gcp.c:68 at wm_gcp_main(): DEBUG: Fetching logs finished.
2021/09/28 13:55:43 wazuh-modulesd:gcp-pubsub[5945] wm_gcp.c:60 at wm_gcp_main(): DEBUG: Sleeping until: 2021/09/28 13:56:23
```

### PR DESCRIPTION
**Pull request by noise-kngdm**
|Related issue|
|---|
|Closes #10334|

## Description
In this PR we modifiy the GCP module to stop using a custom  `<logging>` parameter and use *wazuh_modules* debug value instead, as the other modules currently do. This will ease the task of maintaining all the external integration in an homogeneous way.

Since a parameter was deprecated, several tests had to be removed or modified to adapt them to the state of the module.

Runing valgrind while passing the tests shows the following output:

```
➜  build git:(feature/10334-standardize-gcp-logging) ✗ valgrind ctest --test-dir wazuh_modules/gcp
==245058== Memcheck, a memory error detector
==245058== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==245058== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==245058== Command: ctest --test-dir wazuh_modules/gcp
==245058== 
Internal ctest changing into directory: /home/gonzz/git/wazuh/src/unit_tests/build/wazuh_modules/gcp
Test project /home/gonzz/git/wazuh/src/unit_tests/build/wazuh_modules/gcp
    Start 1: test_wm_gcp
1/2 Test #1: test_wm_gcp ......................   Passed    0.05 sec
    Start 2: test_wmodules_gcp
2/2 Test #2: test_wmodules_gcp ................   Passed    0.07 sec

100% tests passed, 0 tests failed out of 2

Total Test time (real) =   0.39 sec
==245058== 
==245058== HEAP SUMMARY:
==245058==     in use at exit: 0 bytes in 0 blocks
==245058==   total heap usage: 5,061 allocs, 5,061 frees, 1,405,700 bytes allocated
==245058== 
==245058== All heap blocks were freed -- no leaks are possible
==245058== 
==245058== For lists of detected and suppressed errors, rerun with: -s
==245058== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

On the other hand, running pytest shows the following output:

```
(wazuh-unit) ➜  wodles git:(feature/10334-standardize-gcp-logging) ✗ pytest gcloud
==================================== test session starts ====================================
platform linux -- Python 3.10.2, pytest-6.2.5, py-1.11.0, pluggy-0.13.1
rootdir: /home/gonzz/git/wazuh/wodles
plugins: asyncio-0.15.1
collected 17 items                                                                          

gcloud/tests/test_access_logs.py .                                                    [  5%]
gcloud/tests/test_bucket.py ...                                                       [ 23%]
gcloud/tests/test_gcloud.py .....                                                     [ 52%]
gcloud/tests/test_integration.py .....                                                [ 82%]
gcloud/tests/test_subscriber.py ...                                                   [100%]

===================================== warnings summary ======================================
../../../venv/wazuh-unit/lib/python3.10/site-packages/google/pubsub_v1/services/publisher/client.py:17
  /home/gonzz/venv/wazuh-unit/lib/python3.10/site-packages/google/pubsub_v1/services/publisher/client.py:17: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
    from distutils import util

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=============================== 17 passed, 1 warning in 0.07s ===============================
```

### Deprecation messages
When running the module in debug 2 mode,it will output a message saying that the `logging` parameter was deprecated.

#### Pub/Sub

```
2022/03/17 16:34:49 wazuh-modulesd[2083] wmodules-gcp.c:187 at wm_gcp_pubsub_read(): DEBUG: Tag 'logging' from the 'gcp-pubsub' module is deprecated. This setting will be skipped.
2022/03/17 16:34:49 wazuh-modulesd[2083] main.c:94 at main(): DEBUG: Created new thread for the 'gcp-pubsub' module.
...
...
```
#### Buckets
```
2022/03/17 16:36:37 wazuh-modulesd[5147] wmodules-gcp.c:270 at wm_gcp_bucket_read(): DEBUG: Tag 'logging' from the 'gcp-bucket' module is deprecated. This setting will be skipped.
2022/03/17 16:36:37 wazuh-modulesd:gcp-bucket[5147] wmodules-gcp.c:273 at wm_gcp_bucket_read(): DEBUG: Found a bucket tag
...
...
```

### Tests
To test that the modules are working as expected, we performed some manual tests.

#### Manual tests Pub/Sub

The configuration used for the tests is the following:

```
<gcp-pubsub>
    <pull_on_start>yes</pull_on_start>
    <interval>1d</interval>
    <project_id>wazuh-dev-123456</project_id>
    <subscription_name>test_gcloud_blog</subscription_name>
    <logging>debug</logging>
    <credentials_file>/var/ossec/wodles/gcloud/credentials.json</credentials_file>
</gcp-pubsub>
```

##### Test with debug disabled
Only `WARNING`, `ERROR` and `CRITICAL` messages should be captured. We modified the configuration snippet to trigger the error message.

```
2022/03/17 16:50:47 wazuh-modulesd:gcp-pubsub: ERROR: An exception happened while running the wodle: GCloudPubSubSubscriptionError: The 'wazuh-dev-123456' subscription is incorrect or the user credentials are not valid
```

##### Test with debug=1
`INFO` messages should be captured too.

```
2022/03/17 16:57:16 wazuh-modulesd:gcp-pubsub[5220] wm_gcp.c:334 at wm_gcp_pubsub_run(): INFO: Received and acknowledged 0 messages
```

##### Test with debug=2
`DEBUG` messages should be captured too.

```
2022/03/17 16:58:48 wazuh-modulesd:gcp-pubsub[7008] wm_gcp.c:328 at wm_gcp_pubsub_run(): DEBUG: Setting 1 thread to pull 100 messages in total
```

#### Manual tests buckets
The configuration used for the tests is the following:

```
 <gcp-bucket>
    <run_on_start>yes</run_on_start>
    <interval>1m</interval>
    <logging>debug</logging>

    <bucket type="access_logs">
        <name>framework-test-bucket</name>
        <credentials_file>/var/ossec/wodles/gcloud/credentials.json</credentials_file>
        <only_logs_after>2021-oct-01</only_logs_after>
    </bucket>
</gcp-bucket>
```

##### Test with debug disabled
Only `WARNING`, `ERROR` and `CRITICAL` messages should be captured from the Python script. We modified the configuration snippet to trigger the error message.

```
2022/03/17 17:02:38 wazuh-modulesd:gcp-bucket: ERROR: An exception happened while running the wodle: GCloudBucketNotFound: The bucket 'fframework-test-bucket' does not exist
```

##### Test with debug=1
`INFO` messages should be captured too.

```
2022/03/17 17:03:48 wazuh-modulesd:gcp-bucket[11608] wm_gcp.c:439 at wm_gcp_bucket_run(): INFO: Processing test_1_new.txt
2022/03/17 17:03:48 wazuh-modulesd:gcp-bucket[11608] wm_gcp.c:439 at wm_gcp_bucket_run(): INFO: Processing test_22.txt
2022/03/17 17:03:48 wazuh-modulesd:gcp-bucket[11608] wm_gcp.c:439 at wm_gcp_bucket_run(): INFO: Processing test_2_new.txt
```

##### Test with debug=2
`DEBUG` messages should be captured too. Since the buckets don't have any debug message yet, we modified the script adding a test debug message to ensure this was working too.

```
2022/03/17 17:07:15 wazuh-modulesd:gcp-bucket[14839] wm_gcp.c:433 at wm_gcp_bucket_run(): DEBUG: This is a test message
```
#### Test multiline messages 
The C module now supports multiline messages. This is useful when the user faces unknown errors because the full exception trace will be available for both the user and developers to fix the issue. To test this we temporarily modified the `gcloud.py` script to forcefully raise an exception.
    ```
	2022/03/21 12:03:49 wazuh-modulesd:gcp-pubsub[40107] wm_gcp.c:324 at wm_gcp_pubsub_run(): WARNING: Command returned exit code 231
	2022/03/21 12:03:49 wazuh-modulesd:gcp-pubsub[40107] wm_gcp.c:368 at wm_gcp_parse_output(): ERROR: Unknown error
	Traceback (most recent call last):
	  File "/var/ossec/wodles/gcloud/gcloud.py", line 36, in main
		raise TypeError("\nmultiline\nerror\nraise")
	TypeError: 
	multiline
	error
	raised
    ...
    ```

##### Test that the greater log levels include messages from the precedent ones
The different debug levels include the precedent ones, that is, using debug level 1 includes the `WARNING`, `ERROR`, and `CRITICAL` messages that would be returned by debug level 0. Debug level 2 onwards will also include the aforementioned levels, along with the `INFO` -pertaining to debug level 1- and `DEBUG` ones.
###### Debug level 1
To test this we temporarily modified the `gcloud.py` script to forcefully raise an exception. Then we reverted the file to its previous state. To ensure that we would successfully capture a `INFO` log from the Python module.

```
2022/03/21 15:05:03 wazuh-modulesd:gcp-pubsub[59090] wm_gcp.c:324 at wm_gcp_pubsub_run(): WARNING: Command returned exit code 231
2022/03/21 15:05:03 wazuh-modulesd:gcp-pubsub[59090] wm_gcp.c:368 at wm_gcp_parse_output(): ERROR: Unknown error
Traceback (most recent call last):
  File "/var/ossec/wodles/gcloud/gcloud.py", line 37, in main
    raise TypeError("This\nis\nnot\na\ntype\nerror")
TypeError: This
is
not
a
type
error
2022/03/21 15:05:03 wazuh-modulesd:gcp-pubsub[59090] wm_gcp.c:161 at wm_gcp_pubsub_main(): DEBUG: Fetching logs finished.
...
2022/03/21 15:02:14 wazuh-modulesd:gcp-pubsub[57209] wm_gcp.c:362 at wm_gcp_parse_output(): INFO: Received and acknowledged 0 messages
```

###### Debug level 2
For example, executing the module in debug level 2 will return the following output.

    ```
    2022/03/21 12:11:27 wazuh-modulesd:gcp-pubsub[43103] wm_gcp.c:356 at wm_gcp_parse_output(): DEBUG: Setting 10 threads to pull 100 messages in total
    ...
    2022/03/21 12:11:27 wazuh-modulesd:gcp-pubsub[43103] wm_gcp.c:356 at wm_gcp_parse_output(): DEBUG: Sending msg to analysisd: "b'1:Wazuh-GCloud:{"integration": "gcp", "gcp": ...}'"
    ...
    2022/03/21 12:11:27 wazuh-modulesd:gcp-pubsub[43103] wm_gcp.c:362 at wm_gcp_parse_output(): INFO: Received and acknowledged 100 messages

    ```

### CODE REVIEW

File: src/wazuh_modules/wm_gcp.c [hunk 1]
```diff
@@ -312,80 +322,68 @@ void wm_gcp_pubsub_run(const wm_gcp_pubsub *data) {
...
-    }
 
-    os_free(output);
+	parsing_output += cp_length + strlen(WM_GCP_LOGGING_TOKEN) - 1;
+        os_free(tokenized_line);
+    }
 }
 
 void wm_gcp_bucket_run(const wm_gcp_bucket_base *data, wm_gcp_bucket *exec_bucket) {
+    (void) data; // Unused
```
Comment (CarlosRS9) [inline src/wazuh_modules/wm_gcp.c:386]: Why is this unused?

**Review by CarlosRS9 — CHANGES_REQUESTED** [review]

**Review by CarlosRS9 — APPROVED** [review]

File: src/wazuh_modules/wm_gcp.c [hunk 2]
```diff
@@ -312,80 +321,67 @@ void wm_gcp_pubsub_run(const wm_gcp_pubsub *data) {
...
-                    mtinfo(WM_GCP_PUBSUB_LOGTAG, "%s", line);
-                }
-                break;
+    char * parsing_output = output;
+    int debug_level = isDebug();
+
+    for (line = strstr(parsing_output, WM_GCP_LOGGING_TOKEN); line; line = strstr(parsing_output, WM_GCP_LOGGING_TOKEN)) {
+        char * tokenized_line;
+        os_calloc(WM_STRING_MAX, sizeof(char), tokenized_line);
+	char * next_lines;
```
Comment (TomasTurina) [inline src/wazuh_modules/wm_gcp.c:338]: Please, fix line indentation.

File: src/wazuh_modules/wm_gcp.c [hunk 3]
```diff
@@ -312,80 +321,67 @@ void wm_gcp_pubsub_run(const wm_gcp_pubsub *data) {
...
+            }
+            if ((p_line = strstr(tokenized_line, "- WARNING - "))) {
+                p_line += 12;
+                mtwarn(tag, "%s", p_line);
+            }
         }
-    }
 
-    os_free(output);
+	parsing_output += cp_length + strlen(WM_GCP_LOGGING_TOKEN) - 1;
```
Comment (TomasTurina) [inline src/wazuh_modules/wm_gcp.c:379]: Please, fix line indentation.

File: src/wazuh_modules/wm_gcp.c (hunk 2 above)
Comment (noise-kngdm) [reply src/wazuh_modules/wm_gcp.c]: Sorry, my code editor mixed tabs and spaces :angry:

File: src/wazuh_modules/wm_gcp.c [hunk 4]
```diff
@@ -312,80 +321,67 @@ void wm_gcp_pubsub_run(const wm_gcp_pubsub *data) {
...
+        line += strlen(WM_GCP_LOGGING_TOKEN);
+        next_lines = strstr(line, WM_GCP_LOGGING_TOKEN);
+
+        int next_lines_chars = next_lines == NULL ? 0 : strlen(next_lines);
+
+        // 1 is added because it's mandatory to consider the null byte
+        int cp_length = 1 + strlen(line) - next_lines_chars > WM_STRING_MAX ? WM_STRING_MAX : 1 + strlen(line) - next_lines_chars;
+        snprintf(tokenized_line, cp_length, "%s", line);
+        if (tokenized_line[cp_length - 2] == '\n') tokenized_line[cp_length - 2] = '\0'; 
+
```
[the remark this answers is not in the document]
Comment (noise-kngdm) [reply src/wazuh_modules/wm_gcp.c]: Fixed in [664a254](https://github.com/wazuh/wazuh/pull/12769/commits/664a2540494692e73d263a588dc3f1ae1e9681c6).

File: src/wazuh_modules/wm_gcp.c (hunk 3 above)
Comment (noise-kngdm) [reply src/wazuh_modules/wm_gcp.c]: Fixed in [664a254](https://github.com/wazuh/wazuh/pull/12769/commits/664a2540494692e73d263a588dc3f1ae1e9681c6).

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**